### PR TITLE
Internationalization: introduce new linting plugin for i18n errors in blocks

### LINKED ISSUE #15374: Blocks: add @wordpress/eslint-plugin/i18n rules to eslint config
WordPress' `@wordpress/eslint-plugin` will soon (the changes haven't been released yet) include rules to help with internationalization in blocks.

https://github.com/WordPress/gutenberg/pull/20555

Once that's released, I think we should update `extensions/.eslintrc.js` to use the `plugin:@wordpress/eslint-plugin/i18n` configuration.

### PR DESCRIPTION
**Pull request by jeherve**
Fixes #15374

#### Changes proposed in this Pull Request:

* WordPress' `@wordpress/eslint-plugin` plugin includes rules to help with internationalization in blocks; let's use that plugin to help us with block development.

#### Jetpack product discussion

* N/A 

#### Does this pull request change what data or activity we track or use?
* N/A 

#### Testing instructions:

* Do the tests pass? 
* Do the strings with placeholders I edited still appear?

#### Proposed changelog entry for your changes:
* N/A

### CODE REVIEW

File: extensions/blocks/contact-form/edit.js [hunk 1]
```diff
@@ -159,10 +162,7 @@ function JetpackContactFormEdit( {
 			return;
 		}
 
-		const error = e.target.value
-			.split( ',' )
-			.map( validateEmail )
-			.filter( Boolean );
+		const error = e.target.value.split( ',' ).map( validateEmail ).filter( Boolean );
```
Comment (jeherve) [inline extensions/blocks/contact-form/edit.js:165]: This change was made by Prettier.

File: extensions/blocks/podcast-player/utils.js [hunk 2]
```diff
@@ -69,7 +69,6 @@ export function makeCancellable( promise ) {
  * const colors = generateColorsObject( {
  *   primaryColor: '…',
  *   customPrimaryColor: '…',
- *   …
```
Comment (jeherve) [inline extensions/blocks/podcast-player/utils.js:72]: This example was returning an error.

File: extensions/blocks/videopress/edit.js [hunk 3]
```diff
@@ -310,17 +314,19 @@ const VideoPressEdit = CoreVideoEdit =>
...
-								onMouseUp={ this.hideOverlay }
-							/>
-						) }
+							! interactive && (
+								<div
+									className="block-library-embed__interactive-overlay"
+									onMouseUp={ this.hideOverlay }
+								/>
+							)
+						}
```
Comment (jeherve) [inline extensions/blocks/videopress/edit.js:329]: This change was made by Prettier.

File: extensions/shared/blocks/cover/components.js [hunk 4]
```diff
@@ -25,19 +24,20 @@ import UpgradeNudge from '../../components/upgrade-nudge';
  * @returns {*} Nudge component or Null.
  */
 export const JetpackCoverUpgradeNudge = ( { name, show, align } ) =>
-	show
-		? <div className="jetpack-upgrade-nudge-wrapper wp-block" data-align={ align }>
+	show ? (
+		<div className="jetpack-upgrade-nudge-wrapper wp-block" data-align={ align }>
```
Comment (jeherve) [inline extensions/shared/blocks/cover/components.js:28]: This change was made by Prettier.

File: extensions/shared/blocks/cover/components.js [hunk 5]
```diff
@@ -57,8 +57,6 @@ export const CoverMediaContext = createContext();
  */
 export const CoverMediaProvider = ( { onFilesUpload, children } ) => {
 	return (
-		<CoverMediaContext.Provider value={ onFilesUpload }>
-			{ children }
-		</CoverMediaContext.Provider>
+		<CoverMediaContext.Provider value={ onFilesUpload }>{ children }</CoverMediaContext.Provider>
```
Comment (jeherve) [inline extensions/shared/blocks/cover/components.js:60]: This change was made by Prettier.

**Comment by jeherve** [comment]
Putting this on hold, we'll need to land #16534 first.

File: extensions/blocks/repeat-visitor/components/edit.js [hunk 6]
```diff
@@ -42,9 +42,17 @@ class RepeatVisitorEdit extends Component {
...
+				return __(
+					'This block will only appear to people who have visited this page more than once.',
+					'jetpack'
+				);
+			}
+
 			return sprintf(
+				/* translators: placeholder is a number. */
 				_n(
-					'This block will only appear to people who have visited this page more than once.',
```
Comment (zinigor) [inline extensions/blocks/repeat-visitor/components/edit.js:47]: Nice catch!!

**Review by zinigor — APPROVED** [review]
This is awesome, let's get it in!

**Comment by jeherve** [comment]
r219414-wpcom

**Pull request merged**